src/defines.zig

```diff
@@ -81,7 +81,16 @@ pub const DefineData = struct {
                 }
             }
 
-            if (js_lexer.isIdentifier(entry.value_ptr.*) and !js_lexer.Keywords.has(entry.value_ptr.*)) {
+            // check for nested identifiers
+            var valueSplitter = std.mem.split(u8, entry.value_ptr.*, ".");
+            var isIdent = true;
+            while (valueSplitter.next()) |part| {
+                if (!js_lexer.isIdentifier(part) or js_lexer.Keywords.has(part)) {
```

Comment:
What if the value is `foo.class`? `class` is a keyword and its a valid property name, but an invalid identifier.

The first part cannot be a keyword (`import.bar`), but the 2nd one can (`foo.import`)

1. We need to handle the first part differently from the rest.
2. There's probably some edgecases involving spaces or empty parts?

Reply:
For right now, it'll keep the same behavior, which is to cast it as a string. I think it's probably worth considering what the ideal behavior for this is, but it seems safer to increase the scope gradually, rather than allow too much and have to claw it back.

For that reason I decided to compare to what esbuild does. `define`s are validated here:

https://github.com/evanw/esbuild/blob/70893e4d89bab1dbf2d512593f9e4d5bcb1f76bf/pkg/api/api_impl.go#L539

which then calls this function:

https://github.com/evanw/esbuild/blob/70893e4d89bab1dbf2d512593f9e4d5bcb1f76bf/internal/js_parser/js_parser.go#L17348-L17353

which splits on periods and checks that each component is an identifier, without the keyword check.

I think it's probably safe to stick close to what esbuild does, but I don't have strong preferences. I suggest one of the following:

1. check valid identifiers and no keywords - the current code
2. only check for valid identifiers - what esbuild does
3. check all for valid identifiers, and that the first is not a keyword - a split between these, close to what you discuss

do you have a preference among these? Or do you have a case for it being something else?